### PR TITLE
docs: Add "Develop and Deploy your own React monorepo app" blog post

### PR DESCRIPTION
**Pull request by andrico1234**
Adds 'Develop and Deploy your own React monorepo app' post to the Gatsby Blog

Closes https://github.com/gatsbyjs/gatsby/issues/9876

### CODE REVIEW

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+
+The same goes for those who have worked as part of a large company/team. Branding guidelines come in the form of css style-guides or reusable components. Separating the discrete packages from what’s shared becomes a hassle if everything’s stored together. You may even have to deal with several similar projects thrown about in different repos, which can be an arse-ache to keep track of. These are some of the problems monorepos solve.
+
+![Monorepo Diagram](./images/monorepo-diagram.jpeg)
+
+A monorepo is an architectural design pattern that allows us to maintain several packages within a single repo. This reduces the complexity when dealing with packages that share in-house and third-party dependencies.
+
+Think of monorepos like a well-organised supermarket. Each aisle contains a particular kind of food, making it easy to navigate and find the item of food you want. Supermarkets also contain a variety of different foods so you don’t need to go to a series of specific shops to find your gluten-free, dairy-free banoffee pie (which is very much a [thing](https://glutenfreecuppatea.co.uk/2016/02/14/recipe-healthy-banoffee-pie-gluten-free-vegan-refined-sugar-free-dairy-free/)).
```
Comment (shannonbux) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: well I just looked up that pie :)

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 2]
```diff
@@ -0,0 +1,598 @@
...
+
+With monorepos the code is split into specific packages (aisles). Meaning you can easily navigate the project to find the component or module you want to work on. It also means not needing to maintain separate repos when making changes that affect multiple parts of your stack (the single shop).
+
+Everything is accessible from a single place, while still being organised enough to navigate painlessly.
+Enough with the theory, let’s move on to the tutorial, which is split into 2 sections: development and deployment.
+
+Development — We’ll configure an existing application, built with Gastby, into a monorepo. If you haven’t heard of Gatsby before, then take some time to [read up on it](https://www.gatsbyjs.org/). I won’t delve too much into its implementation details because it’s beyond the scope of this article. We’ll attend the lavish party that the great Gatsby throws for us and we won’t ask why or how they did it. Another tool we’ll use is [Lerna](https://github.com/lerna/lerna), which will manage the dependencies of our project. Lerna will also allow us to link packages together which we’ll use to share components.
```
Comment (shannonbux) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: Fun writing details. Love this

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 3]
```diff
@@ -0,0 +1,598 @@
...
+
+Do the same for your shop url, commit and push. Congrats!
+
+### Conclusion
+
+You should now be able to access the site and navigate easily between your shop and your blog. Great job! If things didn’t work out quite as planned then don’t sweat it. This tutorial is made of tiny moving parts which make for loads of opportunities for things to go wrong. I’ll try to make myself available to answer any messages left in the comments regarding any issues you come across.
+
+### Thanks for reading!
+
+Adding to your arsenal of Web Architecture tools is an excellent way of improving your capabilities as a developer or engineer. Monorepos aren’t perfect but it might just work for your next project. Good luck.
```
Comment (shannonbux) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: is web architecture capitalized?

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 4]
```diff
@@ -0,0 +1,598 @@
...
+
+[Add a touch of Suspense to your web app with React.lazy()](https://codeburst.io/add-a-touch-of-suspense-to-your-web-app-with-react-lazy-374e66ee05af)
+
+[No need to wait for the holidays, start Decorating now](https://codeburst.io/no-need-to-wait-for-the-holidays-start-decorating-now-67b9dabd60d7)
+
+[Managing local state with Apollo and Higher Order Components](https://itnext.io/managing-local-state-with-apollo-client-3be522258645)
+
+[The React Conference drinking game](https://medium.com/@andricokaroulla/the-react-conference-drinking-game-7a996bfbef3)
+
+### Troubleshooting
```
Comment (shannonbux) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: Cool to have this section!

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md (hunk 2 above)
Comment (shannonbux) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: Oh, Gatsby is misspelled in this paragraph

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 5]
```diff
@@ -0,0 +1,598 @@
...
+
+## Learn how to create a balanced Lasagna instead of a tangled mess of Spaghetti.
+
+### Foreword
+
+Please put aside time for this tutorial and make sure you’ve got a good hour or two free from distractions. If I’ve done my job right, you’ll improve your general knowledge of web architecture. You’ll also be introduced to new tools whose strengths you can leverage in the wild. If you’ve used any, or all, of these tools before, then getting up and running won’t take as long. But for those unfamiliar with Lerna, Travis or Now, you’ll likely find a few traps along the way, as I did during my journey writing this tutorial. I’ve got a troubleshooting section at the end of the article for common problems, so refer to it at any time. That said, enjoy and I hope you gain a lot from this article.
+
+### Introduction
+
+If you work alone or in a small team, you’re likely to have started working on small programs that slowly grow complex and convoluted. Shit happens and you’ll soon find that your perfectly balanced Lasagna evolves into a heaped bowlful of Spaghetti.
```
Comment (shannonbux) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: are lasagna and spaghetti capitalized for a reason?

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 6]
```diff
@@ -0,0 +1,598 @@
...
+
+Go back to your IDE and you’ll see that we’re not pulling data from an external source. The data is being served locally via a couple of json files in the `data` directory. When we run the `develop` command Gatsby pulls the data from these json files and creates pages for them. You can take a look at how this being done by peeking inside of the `gatsby-node.js` file. This is because Gatsby builds the code and outputs a static site.
+
+So imagine our application is starting to grow, the shop and the blog start intertwining with each other. You may not notice the lack of separation until its too late and your code is tangled tighter than the ivy across the side of your grandma’s cottage.
+
+Let’s start splitting up our application. I’ll only be touching the surface of what Lerna is capable of, so I’d recommend checking out their repo to learn more. Install Lerna with `yarn global add lerna`. We need to create a file in the root directory called `lerna.json`. Add in the following:
+
+```
```
Comment (shannonbux) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: We have some brand new ways of making file names and code types clear

file name using

```shell
```jsx:title=gatsby-config.js
```
at the top of the code snippet. 


type tabs
#10515

Comment (shannonbux) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: I'd recommend this, though it's still optional in the blog. We are requiring it in docs now

Comment (andrico1234) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: thanks, i'll get that in!

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md (hunk 5 above)
Comment (andrico1234) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: yes they are, the reason is that my grammar is bad :)

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md (hunk 4 above)
Comment (andrico1234) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: 💯 💯 💯

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md (hunk 3 above)
Comment (andrico1234) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: I'll change those to small caps!

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md (hunk 2 above)
Comment (andrico1234) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: 😳

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 7]
```diff
@@ -0,0 +1,598 @@
...
+
+This will allow us to transition between our two sites. Do the same for the shop package and you’ll be able to use the site as normal. If `yarn run:all` still works and you can run `yarn build` for each package then commit and push the changes. In each package run `yarn serve` go to the correct port to double-check that everything displays correctly.
+
+Congratulations, the site now behaves in exactly the same way as when we started. This might make it seem like monorepos have little use, but you’ll find that using a monorepo is a way of keeping packages independent from each other while easily reusing assets.
+
+You can stop the tutorial now and you’ll have a working knowedge of when, how and why to use monorepos, but the latter half of this tutorial will show us how we can take things a step further. We’ll get a CI pipeline set up for our application and deploy it on success.
+
+### Part Two: Claire (Deployment)
+
+Let’s take a second to reflect on where we are right now. We’ve managed to start with a single application with two discrete sections and split them apart in a monorepo. This is allows us to _separate the concerns_ and share reusable components.
```
Comment (calcsam) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: this allows us, not "this is allows us"

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md (hunk 5 above)
Comment (calcsam) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: Can we change "Shit" to "Crap"? I don't believe we have curse words on the Gatsby blog.

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 8]
```diff
@@ -0,0 +1,598 @@
...
+
+You should now be able to access the site and navigate easily between your shop and your blog. Great job! If things didn’t work out quite as planned then don’t sweat it. This tutorial is made of tiny moving parts which make for loads of opportunities for things to go wrong. I’ll try to make myself available to answer any messages left in the comments regarding any issues you come across.
+
+### Thanks for reading!
+
+Adding to your arsenal of Web Architecture tools is an excellent way of improving your capabilities as a developer or engineer. Monorepos aren’t perfect but it might just work for your next project. Good luck.
+
+If you enjoyed the article then please reach out to me on [twitter](https://twitter.com/andricokaroulla?lang=en)!
+
+### You can also check out my other articles below:
```
Comment (calcsam) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: Please remove this section, since these other articles are not directly related to the topic at hand.

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md (hunk 3 above)
Comment (calcsam) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: This paragraph is too wordy, please delete

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 9]
```diff
@@ -0,0 +1,598 @@
...
+
+Even if you push everything forward now the site’s navigation will still be broken. The very last thing we need to do is add our alias URL as an environment variables in our `.env.production` file. In your blog package’s `.env.production` go ahead and add the following:
+
+`BLOG_URL=https://<your chosen alias>.now.sh/`
+
+Do the same for your shop url, commit and push. Congrats!
+
+### Conclusion
+
+You should now be able to access the site and navigate easily between your shop and your blog. Great job! If things didn’t work out quite as planned then don’t sweat it. This tutorial is made of tiny moving parts which make for loads of opportunities for things to go wrong. I’ll try to make myself available to answer any messages left in the comments regarding any issues you come across.
```
Comment (calcsam) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]:
```suggestion
 You should now be able to access the site and navigate easily between your shop and your blog. 
```

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 10]
```diff
@@ -0,0 +1,598 @@
...
+
+### Thanks for reading!
+
+Adding to your arsenal of Web Architecture tools is an excellent way of improving your capabilities as a developer or engineer. Monorepos aren’t perfect but it might just work for your next project. Good luck.
+
+If you enjoyed the article then please reach out to me on [twitter](https://twitter.com/andricokaroulla?lang=en)!
```
Comment (calcsam) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]:
```suggestion
 If you enjoyed the article, or have any questions, please reach out to me on [twitter](https://twitter.com/andricokaroulla?lang=en)!
```

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 11]
```diff
@@ -0,0 +1,598 @@
...
+
+Due to the scale of our demo site, you may feel that monorepos aren’t practical. You’ll be right in thinking that because a monorepo in this instance isn’t practical. There’ll be elements of duplication and rewriting that will seem counter productive. It’ll also seem like we’re over-engineering a simple site. The real learning comes from understanding how the pieces fit together on a small scale. You’ll be able to easily integrate monorepos at production level with your understanding of the fundamental concepts and underlying technologies.
+
+This tutorial deals with loads of small changes which make for plenty of opportunities for things to go wrong. I’ve tried making myself as clear as possible to prevent this but if you do come across issues then leave a descriptive commentI’ll do my best to respond, edit and improve this article. You can also find the [finished repo here](https://github.com/andrico1234/lerna-monorepo-finisher). There’s no shame in copying my code if you’re stuck.
+
+I’m also assuming fundamental web dev knowledge including experience with React, Git and the command line.
+
+### Part One: Justine (Development)
```
Comment (calcsam) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: Why do the sections have names? Can we remove them please?

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 12]
```diff
@@ -0,0 +1,598 @@
+---
+title: Develop & Deploy Your Own React Monorepo App in Under 2 Hours, Using Lerna, Travis & Now
```
Comment (calcsam) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]:
```suggestion
 title: Publish Multiple Gatsby Sites in a Monorepo Using Lerna, Travis & Now
```

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 13]
```diff
@@ -0,0 +1,598 @@
...
+date: 2018-12-15
+author: Andrico Karoulla
+tags:
```
Comment (calcsam) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: please make an array

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 14]
```diff
@@ -0,0 +1,598 @@
...
+  - CI
+  - continuous integration
+excerpt: Learn how to create a balanced Lasagna instead of a tangled mess of Spaghetti.
+---
+
+## Learn how to create a balanced Lasagna instead of a tangled mess of Spaghetti.
+
+### Foreword
+
+Please put aside time for this tutorial and make sure you’ve got a good hour or two free from distractions. If I’ve done my job right, you’ll improve your general knowledge of web architecture. You’ll also be introduced to new tools whose strengths you can leverage in the wild. If you’ve used any, or all, of these tools before, then getting up and running won’t take as long. But for those unfamiliar with Lerna, Travis or Now, you’ll likely find a few traps along the way, as I did during my journey writing this tutorial. I’ve got a troubleshooting section at the end of the article for common problems, so refer to it at any time. That said, enjoy and I hope you gain a lot from this article.
```
Comment (calcsam) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]:
```suggestion
 This tutorial will take about an hour to complete. If you're unfamiliar with Lerna, Travis or Now, you may run into a snafu or two, but you'll learn how to use these tools! I’ve included a troubleshooting section at the end of the article for common problems, so refer to it at any time.
```

Comment (calcsam) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: This is less chatty and tighter.

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 15]
```diff
@@ -0,0 +1,598 @@
...
+
+### Introduction
+
+If you work alone or in a small team, you’re likely to have started working on small programs that slowly grow complex and convoluted. Shit happens and you’ll soon find that your perfectly balanced Lasagna evolves into a heaped bowlful of Spaghetti.
+
+Note: I'm running a [Gatsby workshop](https://www.eventbrite.com/e/gatsbyjs-workshop-develop-and-deploy-your-own-blog-tickets-52432622385?aff=GatsbyBlog) in London, February 2019.
```
Comment (calcsam) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: Please include this at the end, not here. Also, please tweet this out and send us a link by DM -- we'll be sure to retweet this!

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 16]
```diff
@@ -0,0 +1,598 @@
...
+
+The same goes for those who have worked as part of a large company/team. Branding guidelines come in the form of css style-guides or reusable components. Separating the discrete packages from what’s shared becomes a hassle if everything’s stored together. You may even have to deal with several similar projects thrown about in different repos, which can be an arse-ache to keep track of. These are some of the problems monorepos solve.
```
Comment (calcsam) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: LOL @ arse-ache, but let's go with "headache"

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md (hunk 2 above)
Comment (calcsam) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: *Development* — We’ll configure an existing application, built with [Gatsby](https://www.gatsbyjs.org/), into a monorepo. 

Another tool we’ll use is [Lerna](https://github.com/lerna/lerna), which will manage the dependencies of our project. Lerna will also allow us to link packages together which we’ll use to share components.

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 17]
```diff
@@ -0,0 +1,598 @@
...
+
+Development — We’ll configure an existing application, built with Gastby, into a monorepo. If you haven’t heard of Gatsby before, then take some time to [read up on it](https://www.gatsbyjs.org/). I won’t delve too much into its implementation details because it’s beyond the scope of this article. We’ll attend the lavish party that the great Gatsby throws for us and we won’t ask why or how they did it. Another tool we’ll use is [Lerna](https://github.com/lerna/lerna), which will manage the dependencies of our project. Lerna will also allow us to link packages together which we’ll use to share components.
+
+Deployment — After we’ve converted our app into a monorepo we’ll hook our GitHub account to Travis-CI and create a CI pipeline. The CI pipeline will test, build and deploy our application. W̵e̵’̵l̵l̵ ̵b̵e̵ ̵u̵s̵i̵n̵g̵ ̵H̵e̵r̵o̵k̵u̵ ̵̵to h̵andle-d̵e̵p̵l̵o̵y̵m̵e̵n̵t̵s ̵t̵o̵o̵l̵ ̵b̵e̵c̵a̵u̵s̵e̵ ̵i̵t̵’̵s̵ ̵e̵a̵s̵y̵ ̵t̵o̵ ̵u̵s̵e̵,̵ ̵f̵l̵e̵x̵i̵b̵l̵e̵ ̵a̵n̵d̵ ̵i̵n̵t̵e̵g̵r̵a̵t̵e̵s̵ ̵r̵e̵a̵l̵l̵y̵ ̵w̵e̵l̵l̵ ̵w̵i̵t̵h̵ ̵T̵r̵a̵v̵i̵s̵-̵C̵I̵.̵ I changed my mind after spending three days trying to get my Monorepo deployed with Heroku, and managed it in an hour with Now. So we’re using Now!
```
Comment (calcsam) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: Please remove quotes. Btw, does this still work with Now v2?

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 18]
```diff
@@ -0,0 +1,598 @@
...
+
+There are a couple of caveats to address prior to beginning:
+
+Due to the scale of our demo site, you may feel that monorepos aren’t practical. You’ll be right in thinking that because a monorepo in this instance isn’t practical. There’ll be elements of duplication and rewriting that will seem counter productive. It’ll also seem like we’re over-engineering a simple site. The real learning comes from understanding how the pieces fit together on a small scale. You’ll be able to easily integrate monorepos at production level with your understanding of the fundamental concepts and underlying technologies.
```
Comment (calcsam) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: Don't write defensively. Instead, say something like?

```suggestion
Due to the scale of our demo site, you may feel that monorepos aren’t practical. You’ll be right in thinking that because a monorepo in this instance isn’t practical. It may be more useful in general, for example if you're creating multiple Gatsby sites in your organization and want to share components easily.
```

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md (hunk 11 above)
Comment (calcsam) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: Part One and Part Two are fine

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 19]
```diff
@@ -0,0 +1,598 @@
...
+`git clone git@github.com:andrico1234/lerna-monorepo-starter.git`
+
+and navigate into the the `lerna-monorepo-starter` directory.
+
+Go to your GitHub account and create a new repo, this’ll be the repo that you push the site to so that Travis can run the pipeline. For those who need a refresher on how to [create a new repo](https://help.github.com/articles/creating-a-new-repository/). You may have to run `git remote remove origin` so you can point the code to your newly created repo.
+
+Run the code that GitHub outputs when creating a new repo:
+
+```
+git remote add origin git@github.com:andrico1234/lerna-monorepo-runthrough.git
```
Comment (calcsam) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: `lerna-monorepo-runthrough` or `lerna-monorepo-starter`?

Please be consistent

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 20]
```diff
@@ -0,0 +1,598 @@
...
+```
+packages
+ | blog
+ | shared-ui
+ | shop
+```
+
+The next step is to move the contents from our current application into their respective folders. All the files in our `components` folder will be moved to our `shared-ui` directory since we’d like to be able to use them anywhere throughout our codebase.
+
+[confirm with run through]We shouldn’t need to reinstall Gatsby to get our shared-ui components working, but we may need to install gatsby-link so the routing doesn’t break.
```
Comment (calcsam) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: please remove note

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md [hunk 21]
```diff
@@ -0,0 +1,598 @@
...
+
+A lot can go wrong in the first half of the application, so make sure that your files sit in their correct place. Also check you’re importing from the `shared-ui` package correctly.
+
+My first go to when the application doesn’t work as expected is to run `lerna clean` and `lerna bootstrap`. This clears all of your `node_modules` and reinstalls them.
+
+You need `react` specified in the package level `package.json` file to run your application. And you’ll also need `react` and `react-router-dom` specified in the top-level `package.json` to run the tests. This left me stumped, and the only logical reason why I think this happens is because Gatsby only seems to look for React in the package level `node_modules` directory, Since the tests are run from the top level, React also needs to exist there.
+
+My suggestion would be to have the package level react dependency the latest version of React, while the top-level react dependency as version 16.0.0.
+
+Testing your application:
```
Comment (calcsam) [inline docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: remove empty sections

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md (hunk 9 above)
Comment (andrico1234) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: why?

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md (hunk 11 above)
Comment (andrico1234) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: A stylistic flourish found in a couple of my articles. I don't see a need to remove them

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md (hunk 14 above)
Comment (andrico1234) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: will condense the foreword a little

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md (hunk 15 above)
Comment (andrico1234) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: can do, will be marketing it a little more in the new year!

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md (hunk 12 above)
Comment (andrico1234) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: I'd like to keep the keywords Develop and Deploy in the title, as it describes the content of my article better, referencing the way in which it's structured. 'Publish Multiple Gatsby Sites' doesn't capture it as much. Saying that, I'm happy to replace 'React' with 'Gatsby' as a means of emphasising our favourite highly performant site generator

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md (hunk 18 above)
Comment (andrico1234) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: I'm not sure how I feel about this. The current paragraph describes, in detail, some of the arguments against using a monorepo for a site this small. I then proceed to provide a counter-argument as to why it would be useful to convert the application over regardless. I'd rather not remove the rationale

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md (hunk 2 above)
Comment (andrico1234) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: I can tighten the paragraph up, but I'd like to still emphasise that this article won't be covering the basics of Gatsby. That's important to see the reader's expection

**Comment by andrico1234** [comment]
@shannonbux @calcsam 

Thank you both for taking the time to read and leave comments.

Shannon - I've resolved all of your comments

Sam - I've resolved most of yours, but there are of comments I don't agree on, happy to discuss them.

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md (hunk 17 above)
Comment (andrico1234) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: this issue gets solved by explicitly stating the version of Now being used from within the `now.json` file

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md (hunk 11 above)
Comment (calcsam) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: This is the Gatsby blog, so we have a particular editorial voice and level of profesisonalism. This is too informal.

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md (hunk 9 above)
Comment (calcsam) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: * No email address or contact info
* Implies the tutorial is buggy
* No comments on this blog

File: docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md (hunk 12 above)
Comment (calcsam) [reply docs/blog/2018-12-15-develop-and-deploy-your-own-react-monorepo-app/index.md]: Title should be more descriptive and a hook to understand what you would do at a glance, and "Publish Multiple Gatsby Sites" is more descriptive than "Develop & Deploy Your Own React[/Gatsby] Monorepo App". 

We don't use the language "Gatsby Monorepo App" anywhere else, and it's not clear immediately what this means and raises additional Qs. Are all Gatsby apps monorepos? Etc.

**Comment by calcsam** [comment]
Thanks for adding the revisions! Responded on those points.

**Comment by andrico1234** [comment]
@shannonbux @calcsam 

made the next set of amendments. Please can you review them and update your reviews accordingly. Thank you!

**Comment by andrico1234** [comment]
@shannonbux @calcsam 

notifying both of you to make you aware that I've made your changes
